jenkinsci/neuvector-vulnerability-scanner-plugin
# Jenkin plugin NeuVector vulnerability Issue

### Summary
1. We let the TLS verification as default behavior, provide a checkbox to disable such behavior.
2. Create the extra Jenkins UI for user to input the server certificate for TLS verification
3. Resolve the vulnerability Issue by Certificate pining and hostname verification, thus we create BouncyCastleTrustStrategy for Certificate checking, and WhitelistedHostnameVerifier for hostname verification.

### Jira Ticket
[NVSHAS_8374](https://jira.suse.com/browse/NVSHAS-8374?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel)

### How to verify
1. Set up your Jenkins server
2. Go to Home page > Manage Jenkins > Configuration, search Input the server certificate for verification
3. input the cert you when you setup the server, then press test connection, make sure it works
<img width="1518" alt="image" src="https://github.com/jenkinsci/neuvector-vulnerability-scanner-plugin/assets/145627854/8f2e30d9-7b02-4225-ab6f-91aabb7a76f1">

4. click checkbox Disable server TLS certificate verification, make sure no matter what text you put inside, as long as the information is correct => it should be able to success.
<img width="1526" alt="image" src="https://github.com/jenkinsci/neuvector-vulnerability-scanner-plugin/assets/145627854/b7234c99-48cf-4390-a6c7-4bb8c04218e0">

### Testing done



```[tasklist]
### Submitter checklist
```
